src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -222,12 +210,17 @@ protected void AssertSelf ()
 				throw new WebException (ex.Message, ex, WebExceptionStatus.ConnectFailure, null);
 			}
 
+			if (httpConnection.DoOutput)
+			{
```

Comment:
Please follow the [Mono Code Style Guidelines](http://www.mono-project.com/community/contributing/coding-guidelines/#style-guidelines) (the opening curly brace should hang on the same line as the conditional expression)